### PR TITLE
fix checkPushPermissions by --insecure

### PR DESCRIPTION
**Pull request by cn-sunletian**
Our registry is harbor by http. When I push image, error occurred on function CheckPushPermissions.
Even add "--insecure" that doesn't work, but add "--insecure-pull" it can pull well.So i find "push.go" an error here. have look at #683

### CODE REVIEW

**Comment by cn-sunletian** [comment]
@SamyCoenen @priyawadhwa @tejal29

**Comment by priyawadhwa** [comment]
Hey @cn-sunletian , looks like a linter error on travis:

```
pkg/executor/push.go:62:19: name.NewInsecureRegistry is deprecated: Use the Insecure Option with NewRegistry instead.  (megacheck)
			newReg, err := name.NewInsecureRegistry(destRef.RegistryStr(), name.WeakValidation)
```

**Comment by cn-sunletian** [comment]
> Hey @cn-sunletian , looks like a linter error on travis:
> 
> ```
> pkg/executor/push.go:62:19: name.NewInsecureRegistry is deprecated: Use the Insecure Option with NewRegistry instead.  (megacheck)
> 			newReg, err := name.NewInsecureRegistry(destRef.RegistryStr(), name.WeakValidation)
> ```

ok i fixed it

**Comment by cn-sunletian** [comment]
@kokoro-team

**Comment by cn-sunletian** [comment]
fix checkPushPermissions by --insecure #683

File: pkg/executor/push.go [hunk 1]
```diff
@@ -58,6 +58,13 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 	checked := map[string]bool{}
 	for _, destination := range opts.Destinations {
 		destRef, err := name.NewTag(destination, name.WeakValidation)
+		if opts.Insecure {
+			newReg, err := name.NewRegistry(destRef.RegistryStr(), name.WeakValidation)
```
Comment (priyawadhwa) [inline pkg/executor/push.go]: I think you want to pass in the insecure option to `name.NewRegistry` -- found [here](https://github.com/google/go-containerregistry/blob/master//pkg/name/options.go#L47) in the go-containerregistry code

**Comment by tejal29** [comment]
Looks like the integration test is failing due to 
```
Building images for Dockerfile Dockerfile_test_copy_bucket
Error building images: Failed to build image gcr.io/kaniko-test/docker-dockerfile_test_arg_multi with docker command "[docker build -t gcr.io/kaniko-test/docker-dockerfile_test_arg_multi -f dockerfiles/Dockerfile_test_arg_multi .]": exit status 1 Sending build context to Docker daemon  117.2kB
```

**Comment by jwidauer** [comment]
As @Dina0312 mentioned: Wouldn't something like this do the same thing, be cleaner and less code?

```
if opts.Insecure {
	destRef, err := name.NewTag(destination, name.WeakValidation, name.Insecure)
}
```

**Comment by cn-sunletian** [comment]
> As @Dina0312 mentioned: Wouldn't something like this do the same thing, be cleaner and less code?
> 
> ```
> if opts.Insecure {
> 	destRef, err = name.NewTag(destination, name.WeakValidation, name.Insecure)
> }
> ```

ok,thanks

File: pkg/executor/push.go [hunk 2]
```diff
@@ -69,6 +69,13 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 	checked := map[string]bool{}
 	for _, destination := range opts.Destinations {
 		destRef, err := name.NewTag(destination, name.WeakValidation)
+		if opts.Insecure || opts.InsecureRegistries.Contains(destination) {
```
Comment (tejal29) [inline pkg/executor/push.go]: shd this be `&&` instead of `||`

**Comment by priyawadhwa** [comment]
Hey @cn-sunletian , this we merged #685 a few weeks ago which should fix this issue. Could you try with `gcr.io/kaniko-project/executor:v0.13.0`? My guess is that this should work.

**Comment by tejal29** [comment]
Closing this since #685 addresses this issue.

**Comment by pvgbabu** [comment]
@priyawadhwa , i have tried using gcr.io/kaniko-project/executor:v0.13.0 but no luck. is there any way
i can pass  '--skip-tls-verify' parameter to  '_generate_kaniko_spec' in https://kubeflow-pipelines.readthedocs.io/en/latest/_modules/kfp/containers/_container_builder.html 
appreciate all your help.

**Comment by pvgbabu** [comment]
@priyawadhwa  where does cluster builder get the docker repository url. I am getting http instead of https causing the error .

**Comment by priyawadhwa** [comment]
Hey @pvgbabu sorry to hear that. Could you open a new issue with the error you're getting?

**Pull request closed without merging**